crates/shared/src/bad_token/token_owner_finder/solvers/solver_api.rs

```diff
@@ -34,7 +34,7 @@ impl TokenOwnerSolverApi for SolverConfiguration {
         self.query().await.map(|token_owner_pairs| {
             token_owner_pairs
                 .into_iter()
-                .filter_map(|(token, owner)| owner.map(|owner| (token, vec![owner])))
+                .filter_map(|(token, owners)| owners.map(|owners| (token, owners)))
```

Comment:
I found it a bit difficult to parse that expression. Not sure if my attempt is any improvement, though. 😄 
```suggestion
                .filter_map(|(token, owners)| Some((token, owners?)))
```

Reply:
Deprecated. Thanks for noticing though.